loggercheck.go

```diff
@@ -25,14 +34,39 @@ func NewAnalyzer() *analysis.Analyzer {
 		Requires: []*analysis.Analyzer{inspect.Analyzer},
 	}
 
-	checkerKeys := strings.Join(loggerCheckersByName.Keys(), ",")
-	a.Flags.Init("loggercheck", flag.ExitOnError)
-	a.Flags.Var(&l.disable, "disable", fmt.Sprintf("comma-separated list of disabled logger checker (%s)", checkerKeys))
+	if !l.disableFlags {
+		defaultFlags(&a.Flags, l)
+	}
+
 	return a
 }
 
+func defaultFlags(fs *flag.FlagSet, l *loggercheck) {
+	checkerKeys := strings.Join(loggerCheckersByName.Keys(), ",")
+	fs.Init("loggercheck", flag.ExitOnError)
+	fs.Var(&l.disable, "disable", fmt.Sprintf("comma-separated list of disabled logger checker (%s)", checkerKeys))
+
+	fs.Func("logger", "add extra logger checker, format: name:packageImport:funcs, "+
```

Comment:
I think config file should be ok. as in [errorcheck](https://github.com/kisielk/errcheck)

Reply:
OK. I've thought about this.
at first I thought a config file is an extra step to user, especially for CI runners.
But I'm OK with the config idea. because the flags seems too long for our use case.

Reply:
refactoring now...

Reply:
the current config file like this:

```yaml
# loggercheck sample config
disable:
    - klog
    - logr
    - zap
custom-checkers:
    - name: mylogger
      package-import: a/customonly
      funcs:
        - (*a/customonly.Logger).Debugw
        - (*a/customonly.Logger).Infow
        - (*a/customonly.Logger).Warnw
        - (*a/customonly.Logger).Errorw
        - (*a/customonly.Logger).With
        - a/customonly.Debugw
        - a/customonly.Infow
        - a/customonly.Warnw
        - a/customonly.Errorw
        - a/customonly.With
  ```

choose yaml because `golangci-lint` also support yaml, and also, yaml is more clear and clean to present list types

Reply:
I'm not sure whether it's better to put all conf in one single yaml file. If users just wanna disable the "klog" checker, it requires them to create a config file, then pass config file as argument to loggercheck, a bit complex.

I think for "custom" rules, one pattern file is enough:
- package imports can be parsed from them
- "Grouping"  can be archived by "comments (# For example)"

Reply:
yes. this is also OK.
but  users just wanna disable the "klog" checker, no need to create config file.
the exists `-disable` flag works as before. and it has higher priority than the config.

Reply:
the whole config idea is intended for things like `golangci-lint`, so they can create a anaylyzer with only one config.
seperate options will make them do extra work, but it is also acceptable

Reply:
I think the most convient use case it using this linter in `golangci-lint`, so when design the feature, 
the first thing I thought is how to make it more easy to intergrate the configable feature to golangci-lint.

Reply:
one pattern file is enough, I do not agree this.

the config file is not only for user, the `Config` struct is also intended to work for tools like golangci-lint so they can use it.

a more general config file, like yaml, will not confuse the user.

strange custom designed single line rule files seems simple but actully it is not. and user need to learn the special format
just like the first flag format I designed